File: java_projects/glue_code/commons-codec/src/main/java/org/apache/commons/codec/digest/Sha2Crypt.java
package org.apache.commons.codec.digest;

import java.util.regex.Pattern;
import org.apache.commons.codec.ContextInitializer;
import org.graalvm.polyglot.Value;

public class Sha2Crypt {
  static final String SHA512_PREFIX = "$6$";
  static final String SHA256_PREFIX = "$5$";
  private static final Pattern SALT_PATTERN =
      Pattern.compile("^\\$([56])\\$(rounds=(\\d+)\\$)?([\\.\\/a-zA-Z0-9]{1,16}).*");
  private static final int SHA512_BLOCKSIZE = 64;
  private static final int SHA256_BLOCKSIZE = 32;
  private static final String ROUNDS_PREFIX = "rounds=";
  private static final int ROUNDS_MIN = 1000;
  private static final int ROUNDS_MAX = 999999999;
  private static final int ROUNDS_DEFAULT = 5000;
  private static Value clz = ContextInitializer.getPythonClass("/digest/Sha2Crypt.py", "Sha2Crypt");
  private Value obj;

  public Sha2Crypt(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/commons-codec/src/main/org/apache/commons/codec/digest/Sha2Crypt.py
from __future__ import annotations
import re
from dataclasses import field
import io


class Sha2Crypt:

    SHA512_PREFIX: str = "$6$"

    SHA256_PREFIX: str = "$5$"
    __SALT_PATTERN: re.Pattern = None  # LLM could not translate this field

    __SHA512_BLOCKSIZE: int = 64

    __SHA256_BLOCKSIZE: int = 32

    __ROUNDS_PREFIX: str = "rounds="

    __ROUNDS_MIN: int = 1000

    __ROUNDS_MAX: int = 999999999

    __ROUNDS_DEFAULT: int = 5000
